solarlog/__init__.py

```diff
@@ -72,7 +77,12 @@ def run(self):
         """
         self.logger.debug("Run method called")
         # setup scheduler for device poll loop
-        self.scheduler_add(self.get_shortname(), self.poll_device, cycle=self.cycle, next=shtime.now())
+        if self.fw:
+            # use 'old' code for firmware <= 2.x
+            self.scheduler_add(self.get_shortname(), self.poll_device, cycle=self.cycle, next=shtime.now())
+        else:
+            # use new code from klab for firmware >= 3.x
+            self.scheduler_add(self.get_shortname(), self.poll_deviceV3, cycle=self.cycle, next=shtime.now())
```

Comment:
Maybe rename the firmware 2 method self.poll_device to self.poll_deviceV2 and assign self.poll_device the correct function?

```
if self.fw == 2:
    # use 'old' code for firmware <= 2.x
    self.poll_device = self.poll_deviceV2
else:
    self.poll_device = self.poll_deviceV3

self.scheduler_add(self.get_shortname(), self.poll_device, cycle=self.cycle, next=shtime.now())
```

Reply:
OK